src/storage/src/hummock/cache.rs

```diff
@@ -677,6 +680,37 @@ impl<K: LruKey, T: LruValue> LruCache<K, T> {
     }
 }
 
+impl<K: LruKey + Clone, T: LruValue> LruCache<K, T> {
+    pub async fn lookup_with_request_dedup<F, VC>(
+        self: &Arc<Self>,
+        hash: u64,
+        key: K,
+        fetch_value: F,
+    ) -> HummockResult<CachableEntry<K, T>>
+    where
+        F: FnOnce() -> VC,
+        VC: Future<Output = HummockResult<(T, usize)>>,
+    {
+        match self.lookup_for_request(hash, key.clone()) {
```

Comment:
Is it possible for `self.lookup_for_request` to take `&K` instead of `K`?

Reply:
No. `lookup_for_request` includes potential insert to a map, and this needs the ownership of `K`.